components/Carousel.tsx

```diff
@@ -0,0 +1,131 @@
+import { useCallback, useRef, useState } from "react";
+import { View, Image, Text, Dimensions, ViewToken } from "react-native";
+import Animated, {
+  Extrapolation,
+  interpolate,
+  SharedValue,
+  useAnimatedScrollHandler,
+  useAnimatedStyle,
+  useSharedValue,
+} from "react-native-reanimated";
+import { CarouselData } from "@/types/carouselData";
+
+const { width } = Dimensions.get("screen");
+const defaultDotWidth = 8;
+
+type CarouselItemProps = {
+  item: CarouselData;
+};
+
+const CarouselItem = ({ item }: CarouselItemProps) => {
+  return (
+    <View
+      className="flex items-center justify-center gap-4 px-4"
+      style={{ width }}
+    >
+      <Image
+        className="h-[200px] w-[300px]"
+        resizeMode="contain"
+        source={item.image}
+      />
+      <Text className="text-center font-clash text-pbk-h5 font-semibold text-white">
```

Comment:
for text related styling I suggest

`pbk-h5` rather than `font-clash text-pbk-h5 font-semibold`